src/EditorFeatures/CSharpTest/GenerateDefaultConstructors/GenerateDefaultConstructorsTests.cs

```diff
@@ -19,62 +19,66 @@ public class GenerateDefaultConstructorsTests : AbstractCSharpCodeActionTest
         protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
             => new GenerateDefaultConstructorsCodeRefactoringProvider();
 
-        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateDefaultConstructors)]
-        public async Task TestProtectedBase()
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateDefaultConstructors)]
+        [InlineData("class")]
+        [InlineData("record")]
+        public async Task TestProtectedBase(string typeKind)
```

Comment:
@CyrusNajmabadi Let me know if you don't want these.

Reply:
not especially.  i'd honestly prefer just a handful of new tests.  maybe a `record Foo;` `record Foo();` and `record Foo { }`  etc. etc.

Reply:
@CyrusNajmabadi There is one problem. For something like the following:

```csharp
record Base(int X)
{
}

record Sub : [||]Base
{
}
```

The constructor added will have parameter name in PascalCase instead of camelCase. Is that acceptable? If not, how to fix that?

Reply:
i'm not understanding what the problem is there.  why would it not be acceptable?

Reply:
@CyrusNajmabadi Because it might be flagged as a naming rule violation. But if that's okay I'll update the test.

Reply:
yeah, i'm still not seeing a problem ehre :)

Reply:
@CyrusNajmabadi It looks like I already done part of this in #50524. One difference between both PRs is that the other PR didn't exclude `IsImplicitlyDeclared` ctors. Should I exclude them or close the PR?

Reply:
> Should I exclude them or close the PR?

what's the difference this makes?  can you give me an example of a scenariot that now works, or a scenario that will now not work?

Reply:
@CyrusNajmabadi For the following code:

![image](https://user-images.githubusercontent.com/31348972/124361629-332c1500-dc30-11eb-8764-b6c2ab27247b.png)

We currently offer to generate two constructors:

![image](https://user-images.githubusercontent.com/31348972/124361637-48a13f00-dc30-11eb-9747-f6585c030706.png)

Excluding `IsImplicitlyDeclared` will only show the user-defined constructor in quick actions. I'm not sure which behavior is the correct one.

Reply:
>Generating the implicitly declared ctor is a valid from language perspective, but is unlikely what the user wants IMO.

let's filter that out hten.

Reply:
@CyrusNajmabadi Can you look at the failing VB test? I think the behavior in test is incorrect for it, but wanted to confirm. Also note that the test name suggests the new behavior 😄 

https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/EditorFeatures/VisualBasicTest/GenerateDefaultConstructors/GenerateDefaultConstructorsTests.vb#L165-L182

The associated internal work item may have some information.